pallets/loans/src/rate_model.rs

```diff
@@ -13,18 +13,27 @@
 // limitations under the License.
 
 use primitives::{Rate, Ratio, Timestamp, SECONDS_PER_YEAR};
-use sp_runtime::traits::{CheckedAdd, CheckedDiv, CheckedSub, Saturating};
+use sp_runtime::traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, Saturating};
 
 use crate::*;
 
+pub trait InterestRateModel {
+    /// Check the model for sanity
+    fn check_model(&self) -> bool;
+
+    /// Calculates the borrow interest rate of the model
+    fn get_borrow_rate(&self, utilization: Ratio) -> Option<Rate>;
+}
+
 /// Parallel interest rate model
 #[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug)]
-pub enum InterestRateModel {
-    Jump(JumpModel),
+pub enum InterestRateModels {
     Curve(CurveModel),
+    Jump(JumpModel),
+    NewCurveName(NewCurveNameModel),
```

Comment:
It is necessary to replace `NewCurveName` with the actual name of the new model (Couldn't figure out one)

Reply:
I think we can call it **PolynomialCurve** since the new curve has more than two algebraic terms, and it is the sum of several terms that contain different powers of the same variable. @c410-f3r

Reply:
Thank you